site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -4,44 +4,300 @@ title: Upgrade FusionAuth
 
-To upgrade to the latest version of FusionAuth using Homebrew go link:/docs/v1/tech/installation-guide/homebrew[here].
+=== Connected Applications and Client Libraries
 
-To upgrade to the latest version of FusionAuth using Docker go link:/docs/v1/tech/installation-guide/docker[here].
+Consider that a FusionAuth server upgrade might mean that you need to upgrade your client libraries too. Check if this is needed before upgrading, as it will entail a more complex upgrade process and may require more downtime.
 
-If you want to use manual mode, download the latest version of FusionAuth and follow the instructions below. See https://fusionauth.io/download[Downloads].
+In general, client libraries are forward-compatible, but any issues will be noted in the release notes. (Read the release notes carefully.)
+
+=== Skipping Versions 
+
+Sometimes, you will want to upgrade to a FusionAuth version that is a few versions newer than your current version. This is called a "version jump". Version jumps include minor versions. FusionAuth migration can run all the necessary database migrations to get you to the latest version. 
+
+In some cases, version jumps are safe to do. To determine if you can perform a version jump, you should review the link:/docs/v1/tech/release-notes[release notes] carefully. If the releases focus on bug fixes and minor improvements, you can probably perform a version jump safely. Additionally, if the only changes are to features that you are not using, you can probably perform a version jump.
+
+Whichever approach you take, you should always test the upgrade in a staging environment to identify potential issues. This is particularly important on more complex upgrades involving multiple version jumps or if you have made customizations to FusionAuth. We recommend that the staging environment be as close as possible to your production environment. This includes using the same data, or at least the same data volume, as production. This enables you to test both functionality changes and data update performance times before committing to the upgrade on production.
+
+=== Migrating Configuration Between Environments
+
+There are a number of options for managing the configuration of your FusionAuth instance over time and promoting changes from one environment to another. These include:
+
+- Script the changes.
+- Terraform.
+- Pulumi.
+- Manual configuration via the UI.
+
+You can read more in link:/docs/v1/tech/admin-guide/configuration-management[the Configuration Management guide].
+
+=== Themes and Customizations
+
+Review custom themes and how they will be impacted by any FusionAuth updates. If you have made any customizations, you may need to update them to work with the new version of FusionAuth. You can use the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] project to perform diffs on the themes to identify any changes that need to be made.
+
+include::docs/v1/tech/installation-guide/_theme-upgrade.adoc[]
```

Comment:
Can you add this into the themes section: https://fusionauth.io/docs/v1/tech/themes/#upgrading

Reply:
the include is already added to that section so information will update after merge